src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -1,275 +0,0 @@
-﻿namespace NServiceBus.Features
-{
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Threading.Tasks;
-    using ConsistencyGuarantees;
-    using DeliveryConstraints;
-    using Extensibility;
-    using Logging;
-    using NServiceBus.DataBus;
-    using NServiceBus.Gateway;
-    using NServiceBus.Gateway.Channels;
-    using NServiceBus.Gateway.Channels.Http;
-    using NServiceBus.Gateway.HeaderManagement;
-    using NServiceBus.Gateway.Installer;
-    using NServiceBus.Gateway.Notifications;
-    using NServiceBus.Gateway.Receiving;
-    using NServiceBus.Gateway.Routing.Endpoints;
-    using NServiceBus.Gateway.Routing.Sites;
-    using NServiceBus.Gateway.Sending;
-    using Performance.TimeToBeReceived;
-    using Routing;
-    using Settings;
-    using Transport;
-
-    /// <summary>
-    /// Used to configure the gateway.
-    /// </summary>
-    [ObsoleteEx(
-        RemoveInVersion = "4.0",
-        TreatAsErrorFromVersion = "3.0",
-        Message = "Use `EndpointConfiguration.Gateway() to enable the gateway.")]
-    public class Gateway : Feature
```

Comment:
I don't think this is supposed to be removed completely, the feature should only be made private